### PR TITLE
Improves Variables usage description in several sections

### PR DESCRIPTION
**Pull request by SrNetoChan**

### CODE REVIEW

File: source/docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -1688,19 +1690,31 @@ Will render the label like this::
...
+values are emphasized in italic. On the other hand, higher level
+variables overwritten by lower level ones are strike through.
 
 .. _figure_variables_dialog:
 
 .. figure:: /static/user_manual/introduction/options_variables.png
    :align: center
 
-   Edit variable at the project level
+   Editable variables at the project's level
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: Doesn't it sound like the displayed variables in the screenshot can be edited (while not all of them are)?

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 2]
```diff
@@ -36,7 +36,7 @@ require a restart of QGIS before they will be effective.
 The tabs where you can customize your options are described below.
 
 General Tab
--------------
+-----------
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: Is this an issue or not at your taste? Because I use to over underline (one or two characters more) when adding new sections, so that I adapt

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 3]
```diff
@@ -525,12 +523,21 @@ http://doc.qt.io/qt-4.8/qnetworkproxy.html#ProxyType-enum
...
 Variables Tab
---------------
+-------------
+
+The :guilabel:`Variables` tab lists all the variables available at the
+global-level.
 
-The :guilabel:`Variables` tab lists and helps you create global-level variables.
-More information about variables at section :ref:`general_tools_variables`.
+It also allows the user to manage global-level variables. Click the |signplus|
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: actually, it's [|signPlus|](http://docs.qgis.org/testing/en/docs/documentation_guidelines/substitutions.html#other-basic-icons) and |signMinus| (I'm not sure this is case insensitive and I'm a bit concerned that Travis does not catch this issue. It's not [the first time](https://github.com/qgis/QGIS-Documentation/pull/1630#issuecomment-273822809).
Can someone please check whether configurations (or anything else) are still valid and applied?

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -2337,11 +2337,28 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
    menu of the layer in :guilabel:`Layers Panel` to automatically get different
    representations of your data.
 
+.. index:: Variables
+.. _vectorvariables_menu:
+
+Variables Properties
+====================
+
+The :guilabel:`Variables` tab lists all the variables available at
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: add variable icon, as done for other sections?

**Comment by DelazJ** [comment]
I added few comments. Otherwise it's very good and covers all the needed points I can see. Thanks @SrNetoChan

File: source/docs/user_manual/introduction/general_tools.rst (hunk 1 above)
Comment (SrNetoChan) [inline source/docs/user_manual/introduction/general_tools.rst]: You are right. "Variables editor at the project level"?

File: source/docs/user_manual/introduction/qgis_configuration.rst (hunk 2 above)
Comment (SrNetoChan) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: I thought this was an issue. Now that you mention, Sphinx throws a warning when the characters are less than the title, but not when they are over. On the other hand, PyCharm complaints about it, saying that the underline must match the title.

I don't have a strong opinion about this, although I prefer to see the underline matching. But, if it's useful for you to keep the extras one or two underlines, I won't change them anymore.

File: source/docs/user_manual/introduction/qgis_configuration.rst (hunk 3 above)
Comment (SrNetoChan) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: It seems to be case insensitive. At least I was able to render the buttons correctly. But I will change it to the official name.

File: source/docs/user_manual/introduction/qgis_configuration.rst (hunk 2 above)
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: Let's make PyCharm happy :)

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 4 above)
Comment (SrNetoChan) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Sure! Missed that.

File: source/docs/user_manual/introduction/qgis_configuration.rst (hunk 3 above)
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: ah, ok! Never notice that.
Yes, let's use a single spelling.

File: source/docs/user_manual/print_composer/overview_composer.rst [hunk 5]
```diff
@@ -449,6 +450,28 @@ and color. These options are applied by default to any new print composer.
 
 .. index:: Revert layout actions
```
Comment (DelazJ) [inline source/docs/user_manual/print_composer/overview_composer.rst]: This index is linked to the Revert actions section

**Comment by DelazJ** [comment]
Actually I miss a description: variable at print composer items (scalebar, legend, map...) level

File: source/docs/user_manual/print_composer/overview_composer.rst [hunk 6]
```diff
@@ -359,6 +359,7 @@ To maximise the space available to interact with a composition you can use
    interact by pressing :kbd:`F11` or using :menuselection:`View -->` |checkbox|
    :guilabel:`Toggle full screen`.
 
+.. _composer_composition_tab:
 
 Composition Tab
 ---------------
```
Comment (DelazJ) [inline source/docs/user_manual/print_composer/overview_composer.rst]: It's not in the scope of this PR but wonder if this shouldn't be labeled as panel rather than tab

Comment (SrNetoChan) [inline source/docs/user_manual/print_composer/overview_composer.rst]: You are indeed panels but are usually grouped in tabs, that why it still makes sense. +0 for changing it to panels.

File: source/conf.py [hunk 7]
```diff
@@ -1200,6 +1200,8 @@
 .. |largeLandUseArea| replace:: Bontebok National Park
 .. |srtmFileName| replace:: :kbd:`srtm_41_19.tif`
 .. |localCRS| replace:: :kbd:`WGS 84 / UTM 34S`
+.. |variables| image:: /static/common/mIconExpression.png
```
Comment (DelazJ) [inline source/conf.py]: There's a [replacement set](http://docs.qgis.org/testing/en/docs/documentation_guidelines/substitutions.html#selection-and-expressions) for this icon(https://github.com/SrNetoChan/QGIS-Documentation/blob/607bd1717ad1a0a7921d31359b084efec3c199fd/source/conf.py#L516)

Comment (SrNetoChan) [inline source/conf.py]: Yes, I noticed that, But since it's not specific to variables, I was wondering if in the future we should propose to change the icon in the future.

**Comment by SrNetoChan** [comment]
@DelazJ Added a description for composer items, ended up by changing the general tools also, If you have time, please review.

File: source/docs/user_manual/print_composer/overview_composer.rst (hunk 6 above)
Comment (DelazJ) [inline source/docs/user_manual/print_composer/overview_composer.rst]: > You are indeed panels but are usually grouped in tabs

Hmm.. Like most of the panels in map canvas. I guess most of the users have them anchored and  tabbed. But never mind, no strong opinion, there!

File: source/conf.py (hunk 7 above)
Comment (DelazJ) [inline source/conf.py]: Currently, there's only one place in the application where "variables" term is associated to an icon and it's as a tab of layer properties dialog ie, here in the doc. So no other place in doc is concerned by the use of any variable replacement, yet. I'm not sure it's  worth a particular replacement, and in case of new icon, the committer will manage the changes when time comes.
Also I think common icon replacement are supposed to be listed altogether (in the list below, in the conf file)

File: source/docs/user_manual/introduction/general_tools.rst [hunk 8]
```diff
@@ -1691,13 +1691,17 @@ Will render the label like this::
...
+
+* **global variables** from the :menuselection:`Settings --> Options` menu;
+* **project's variables** from :guilabel:`Project properties` (see
+  :ref:`project_properties`);
+* **vector layer's variables** from the :guilabel:`Layer Properties` dialog
+  (see :ref:`vector_properties_dialog`);
+* **composition's variables** from the :guilabel:`Composition` tab in the
+  :guilabel:`Map composer` (see :ref:`composer_composition_tab`);
+* and **composer item's variables** from the :guilabel:`Item` tab in the
+  :guilabel:`Map composer` (see :ref:`composer_items_tab`).
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: Not sure to follow given that no variable section is available under Item**s** tab.
Also what does :guilabel:`Map composer` refer to? :guilabel: is supposed to indicate a label available in the gui and I can't find such text in the GUI within print composer

Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: Then I'm not sure about the link in reference

File: source/docs/user_manual/print_composer/composer_items/composer_items_options.rst [hunk 9]
```diff
@@ -150,3 +150,18 @@ matching alignment icon. All selected elements will then be aligned to their com
 When moving items on the Composer canvas, alignment helper lines appear when borders, centers or
 corners are aligned.
 
+Variables
+---------
+
+The :guilabel:`Variables` lists all the variables available at
+the composer item's level (which includes all global, project and
+composition's variables). Map items also includes Map settings variables that
```
Comment (DelazJ) [inline source/docs/user_manual/print_composer/composer_items/composer_items_options.rst]: include

File: source/docs/user_manual/print_composer/composer_items/composer_items_options.rst [hunk 10]
```diff
@@ -150,3 +150,18 @@ matching alignment icon. All selected elements will then be aligned to their com
...
+provide easy access to values like the map's scale, extension, and so on.
```
Comment (DelazJ) [inline source/docs/user_manual/print_composer/composer_items/composer_items_options.rst]: extent?

File: source/conf.py (hunk 7 above)
Comment (SrNetoChan) [inline source/conf.py]: I will change this to use the expressions replacement.

File: source/docs/user_manual/introduction/general_tools.rst (hunk 8 above)
Comment (SrNetoChan) [inline source/docs/user_manual/introduction/general_tools.rst]: "Map composer" should be "Print Composer". Will change that
You are right, I used the wrong reference for the items, it should point to composer_item_options instead.

File: source/docs/user_manual/print_composer/composer_items/composer_items_options.rst (hunk 10 above)
Comment (SrNetoChan) [inline source/docs/user_manual/print_composer/composer_items/composer_items_options.rst]: yup!

File: source/docs/user_manual/print_composer/overview_composer.rst (hunk 6 above)
Comment (SrNetoChan) [inline source/docs/user_manual/print_composer/overview_composer.rst]: Let's do this in another PR. It makes sense to call them panels as we can only turn them on and off using View --> Panels. I opened a new issue for this.

https://github.com/qgis/QGIS-Documentation/issues/1716

Comment (DelazJ) [inline source/docs/user_manual/print_composer/overview_composer.rst]: Thanks for the check!

**Comment by SrNetoChan** [comment]
@DelazJ Everything should be fixed now. Thanks for the review.

File: source/docs/user_manual/print_composer/overview_composer.rst [hunk 11]
```diff
@@ -157,9 +157,12 @@ and :guilabel:`Atlas generation`.
   item. Click the |select| :sup:`Select/Move item` icon to select
   an item (e.g., legend, scale bar or label) on the canvas. Then click the
   :guilabel:`Item Properties` tab and customize the settings for the selected
-  item.
+  item (see :ref:`composer_items` for detailed information on each item
```
Comment (SrNetoChan) [inline source/docs/user_manual/print_composer/overview_composer.rst]: @DelazJ Added this references to the item properties tab and atlas generation tab.
